api/request.go

```diff
@@ -0,0 +1,44 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+	"k8s.io/utils/strings/slices"
+
+	"clouditor.io/clouditor/internal/util"
+)
+
+var (
+	ErrInvalidColumnName = errors.New("column name is invalid")
+	ErrRequestIsNil      = errors.New("request is empty")
+)
+
+// ListRequest indicates a proto message being a request for ListXXX RPCs
+type ListRequest interface {
```

Comment:
I am torn whether we should have interfaces for the messages or whether we should use protoreflect to retrieve these values (we do it this way in the pagination request handling)